erizo/src/erizo/WebRtcConnection.cpp

```diff
@@ -222,46 +239,82 @@ bool WebRtcConnection::werePacketLossesRecently() {
 boost::future<void> WebRtcConnection::addMediaStream(std::shared_ptr<MediaStream> media_stream) {
   return asyncTask([media_stream] (std::shared_ptr<WebRtcConnection> connection) {
     boost::mutex::scoped_lock lock(connection->update_state_mutex_);
-    ELOG_DEBUG("%s message: Adding mediaStream, id: %s", connection->toLog(), media_stream->getId().c_str());
-    connection->media_streams_.push_back(media_stream);
+    ELOG_DEBUG("%s message: Adding mediaStream, id: %s, transceivers_count: %d",
+      connection->toLog(), media_stream->getId().c_str(), connection->transceivers_.size());
+    bool added = false;
+    std::for_each(connection->transceivers_.begin(), connection->transceivers_.end(),
+        [&media_stream, &added, connection] (const auto &transceiver) {
+      if (!transceiver->hasMediaStream() && !added) {
+        transceiver->setMediaStream(media_stream);
+        added = true;
+        ELOG_DEBUG("%s message: Added mediaStream reusing transceiver, id: %s, transceiverId: $d",
+          connection->toLog(), media_stream->getId().c_str(), transceiver->getId());
+      }
+    });
+    if (!added) {
+      auto transceiver = std::make_shared<Transceiver>(std::to_string(connection->transceivers_.size()));
```

Comment:
I think this is the way Chrome sets the ids now. I'm fine with it until we find out if a more sophisticated way to set id's is worth it 👍